cherab/tools/observers/group/base.py

```diff
@@ -83,420 +78,257 @@ def __getitem__(self, item):
+                                 "mismatches the number of sight-lines ({}).".format(len(value), len(self._observers)))
         else:
-            for sight_line in self._sight_lines:
-                sight_line.pixel_samples = value
+            for observer in self._observers:
+                observer.pixel_samples = value
 
     @property
     def samples_per_task(self):
         # Minimum number of samples to request per task.
-        return [sight_line.samples_per_task for sight_line in self._sight_lines]
+        return [observer.samples_per_task for observer in self._observers]
 
     @samples_per_task.setter
     def samples_per_task(self, value):
         if isinstance(value, (list, tuple, ndarray)):
-            if len(value) == len(self._sight_lines):
-                for sight_line, v in zip(self._sight_lines, value):
-                    sight_line.samples_per_task = v
+            if len(value) == len(self._observers):
+                for observer, v in zip(self._observers, value):
+                    observer.samples_per_task = v
             else:
                 raise ValueError("The length of 'samples_per_task' ({}) "
-                                 "mismatches the number of sight-lines ({}).".format(len(value), len(self._sight_lines)))
+                                 "mismatches the number of sight-lines ({}).".format(len(value), len(self._observers)))
         else:
-            for sight_line in self._sight_lines:
-                sight_line.samples_per_task = value
-
-    @property
-    def pipelines(self):
-        # A list of all pipelines connected to each sight-line in the group.
-        return [sight_line.pipelines for sight_line in self._sight_lines]
-
-    def connect_pipelines(self, properties=[(SpectralRadiancePipeline0D, None, None)]):
-        """
-        Connects pipelines of given kinds and names to each sight-line in the group.
-        Connected pipelines are non-accumulating by default.
-
-        :param list properties: 3-tuple list of pipeline properties in order (class, name, filter).
-                                Default is [(SpectralRadiancePipeline0D, None, None)].
-                                The following pipeline classes are supported:
-                                    SpectralRadiacnePipeline0D,
-                                    SpectralPowerPipeline0D,
-                                    RadiacnePipeline0D,
-                                    PowerPipeline0D.
-                                Filters are applied to the mono pipelines only, namely,
-                                PowerPipeline0D or RadiacnePipeline0D. The values provided for spectral
-                                pipelines will be ignored. The filter must be an instance of
-                                SpectralFunction or None.
-
-        """
-
-        for sight_line in self._sight_lines:
-            sight_line.connect_pipelines(properties)
-
```

Comment:
The pipelines property should definitely be included. I think that the readability of the connect_pipelines could be improved. Should I implement it "as is" or should I try to improve it?

Reply:
Yes, please improve this. Probably, `connect_pipelines()` should take a list of (`pipeline_class`, `pipeline_kwargs`) tuples. 

When doing the group observation, it will make sense having `display_progress` set to False in the default kwargs for the spectral pipelines.

Reply:
This seems as reasonable approach. But I am wondering if it would be possible to create interface so that group would be compatible with 1D pipelines, because the most common use case seems to be to have observers with the same pipelines.

Reply:
It is possible but you'll need to completely rewrite the `observe()` method. I'm not sure if this is a good approach. First, the current implementation allows different observers in the group have different spectral settings. If all 0D observers share the same 1D pipeline, this would not be possible. Second, I think this breaks the integrity. For example, you'll loose the observed frame, if you add a new observer to the group between observations even if the pipeline is in accumulation mode. 

However, if it is OK that all observers in a group have the same spectral properties and differ only in geometry, then the group observer can be implemented as `Observer1D`. I think it's easier than sharing 1D pipeline between multiple 0D observers. The disadvantage of this approach is that 1D observers do not have a `samples_per_task` property. Thus, if `MulticoreEngine` is used and the number of observers in the group is less than the number of parallel processes, rendering will not be efficiently parallelized.  Let's hear what @Mateasek, @jacklovell and @CnlPepper have to say about this.

Reply:
Actually, choosing between keeping as is or converting to `Observer1D`, I prefer to leave it as is.

Reply:
I see, let's keep it as it is now.